app/assets/javascripts/actions/index.coffee

```diff
@@ -0,0 +1,37 @@
+Screensmart.Actions =
+  addMessage: (message) ->
+    type: 'ADD_MESSAGE'
+    message: message
+
+  setAnswer: (key, value) ->
+    (dispatch) =>
+      dispatch @_setAnswer(key, value)
+      dispatch @updateResponse()
+
+  _setAnswer: (key, value) ->
+    type: 'SET_ANSWER'
+    key: key
+    value: value
+
+  updateResponse: ->
+    (dispatch, getState) ->
+      response = getState().response
+      syncResponse(response).then (data) ->
+        dispatch Screensmart.Actions.receiveResponseUpdate(data)
+
+  setResponseLoading: ->
+    type: 'SET_RESPONSE_LOADING'
+
+  receiveResponseUpdate: (data) ->
+    type: 'RECEIVE_RESPONSE_UPDATE'
+    response: data.response
+
+syncResponse = (response) ->
```

Comment:
Dan zou ik er `PATCH /responses/:id/sync` van maken. Verdere discussie kan in [dit ticket](https://www.pivotaltracker.com/story/show/120097675)